### ISSUE TITLE
 Add support for rate limiting

### ISSUE THREAD

**Issue by steve-fraser** [linkerd/linkerd2#4649]
## Feature Request
It would be nice if linkerd had rate limiting.
### What problem are you trying to solve?
We have problem where an application gets bombarded by requests if all the clients disconnect. I would like to be able to protect applications through rate limiting.

**Comment by grampelberg** [comment]
@steve-fraser in my experience, folks want to do rate limiting most often at the ingress level. Why are you looking for it on service-service traffic?

**Comment by ltagliamonte-dd** [comment]
Service to service rate limiting is a pattern mostly used to avoid to tip over a service that is temporary under high pressure, would be a great features to have.

**Comment by bsmithyman** [comment]
Just commenting here as we are investigating LinkerD and looking at the feature set available. One of the things that would be wonderful to have in rate-limiting support is the ability to control the rate of outgoing requests from a service at the mesh level.

The difficulty is that during a scaling event, managing the request flow out of a service depends on knowing the number of live replicas of the service communicating with a particular endpoint. We run into this with legacy services, where their ability to handle requests is outpaced by modern Kubernetes-hosted workloads. When we horizontally scale the client workloads, it's quite a complex challenge to account for the pressure we put on legacy backends from within the (stateless) application. Given legacy, sometimes limiting traffic on the layers guarding the slow systems is much more difficult.

The tldr here is:
Rate limiting is particularly useful when modern, mesh-hosted workloads are interacting with workloads outside of the cluster. If this ends up implemented, it might be something to consider in the feature set.

**Comment by wmorgan** [comment]
See #7278
